test/service/test_signals.py

```diff
@@ -32,6 +35,62 @@ def original_name(self):
     def signal_disabled(self):
         assert type(self) is ExampleInterface
 
+    @dbus_property(access=PropertyAccess.READ)
+    def test_prop(self) -> 'i':
+        return 42
+
+
+class SecondExampleInterface(ServiceInterface):
+    def __init__(self, name):
+        super().__init__(name)
+
+    @dbus_property(access=PropertyAccess.READ)
+    def str_prop(self) -> 's':
+        return "abc"
+
+    @dbus_property(access=PropertyAccess.READ)
+    def list_prop(self) -> 'ai':
+        return [1, 2, 3]
+
+
+class ExpectMessage:
+    def __init__(self, bus1, bus2, timeout=1):
+        self.future = asyncio.get_event_loop().create_future()
+        self.bus1 = bus1
+        self.bus2 = bus2
+        self.timeout = timeout
+        self.timeout_task = None
+
+    def message_handler(self, msg):
+        self.timeout_task.cancel()
+        self.bus2.remove_message_handler(self.message_handler)
+        self.future.set_result(msg)
+        return True
+
+    def timeout_cb(self):
+        self.bus2.remove_message_handler(self.message_handler)
+        self.future.set_result(TimeoutError())
+
+    async def __aenter__(self):
+        self.bus2.add_message_handler(self.message_handler)
+        self.timeout_task = asyncio.get_event_loop().call_later(self.timeout, self.timeout_cb)
+
+        return self.future
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        pass
```

Comment:
You need to `await` for `self.future` here (possibly with a timeout via `asyncio.wait_for()`), otherwise the context manager doesn't check anything

Reply:
That is not necessary. We return the future from __aenter__ and await it outside. I however added the filters back to make sure not just any message can trigger the handler.